controllers/drplacementcontrol_controller.go

```diff
@@ -1118,23 +1123,22 @@ func (r *DRPlacementControlReconciler) getDRPCPlacementRule(ctx context.Context,
 	return nil
 }
 
-func (r *DRPlacementControlReconciler) ownPlacementOrPlacementRule(
+func (r *DRPlacementControlReconciler) setDRPCOwnerAndAnnotatePlacement(
 	ctx context.Context,
 	drpc *rmn.DRPlacementControl,
+	usrPlacement client.Object,
 	log logr.Logger,
-) (client.Object, error) {
-	usrPlacement, err := getPlacementOrPlacementRule(ctx, r.Client, drpc, log)
+) error {
+	err := ctrl.SetControllerReference(usrPlacement, drpc, r.Client.Scheme())
```

Comment:
We are setting the controller reference but not calling an `Update`, so when we get to update DRPC status this reference would possibly get added (need to check though).

Should we on initial reconcile just set the controller reference and requeue, and in a future recconcile we check if we already have the ref and proceed?  That way intermediate delete operations would be cleaner, also we avoid updating the current resource under reconcile twice.

Reply:
So this owner ref should get updated on DRPC once we add the Finalizer in CreateDRPCInstance (which has the Update call if we just added a finalizer). For pre-existing DRPC this owner ref may hence not be updated? (just thinking aloud at present)

Reply:
> 1. GetPlacementObj
> 2. AnnotatePlacement (make sure this DRPC is the sole reconciler for the placement)
> 3. SetDRPCOwner
>      - If set for this instance of the call, requeue the reconcile
>.     - If already set, continue reconciliation
> IOW switch 2 and 3, and add a requeue if 3 was performed.

Yes, we can reorder as suggested

Reply:
It should be updated only once. Relying on the first time we update labels or finalizer.